### DISCUSSION TITLE
queue:retry --pattern=Exception\RuntimeException

### DISCUSSION THREAD

**Discussion by nandi95** [laravel/framework#44824] [Ideas]
When there are various failed jobs eg.: email failed to send
it would be nice if I could retry only a certain type of jobs

On the `failed_jobs` there's an `exception` column includes the whole stack trace something like
```
Symfony\Component\Mailer\Exception\TransportException: Expected response code "250/251/252" but got code "550", with message "550 5.7.1 Relaying denied". in /var/task/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php:316
Stack trace:
#0 /var/task/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(180): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->assertResponseCode('550 5.7.1 Relay...', Array)
.
.
.
```

I propose we could include a new argument `--pattern` to which you would normally pass a namespaced error such as `Symfony\Component\Mailer\Exception\TransportException`
This would run all jobs where the exception is born out this error

**Comment by ohnotnow** [comment]
I quite like the idea behind this.  Sometimes I'll have a set of jobs fail for the same reason (API is down, whatever) and it would be handy to be able to retry those jobs in one go without having to dig out their ID's or click them in Horizon.

Would it be worth a confirmation step that showed what was going to be retried?  Just in case of a typo or over-broad 'pattern'?  List of jobs and a y/n prompt?

**Comment by rmcdaniel** [comment]
You could tell the job to retry forever with an exponential backoff. When the API comes back online, the job will be retried and succeed.

**Reply by nandi95** [reply]
That's a solution but not really a good one in my eyes.
Let's say the job fails rightly so because you forgot to do something manually. Why waste resources when we know the re-runs will never succeed until we did the manual action.

**Reply by rmcdaniel** [reply]
If the job fails because your code has a bug then you just fix the bug and restart your queue. The job will restart with the new code and then succeed. I actually made a demo showing this recently using my library which is just a wrapper around queues with some helper stuff to make things easier https://www.youtube.com/watch?v=TnV_liz7cpA but it applies equally to vanilla queues.

**Comment by nandi95** [comment]
Taylor didn't wanna merge because it's a clutter https://github.com/laravel/framework/pull/49072 so that answers it
¯\\_(ツ)_/¯
closing
